aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterClearTest.java

```diff
@@ -0,0 +1,210 @@
+        //Needed so we can make assertions on the state of the subscriber later.
+        observer = adapter
+            .observe()
+            .doOnSubscribe(subscriberDisposableRef::set)
+            .test();
+
+        fileObserver = new TestFileObserver(Objects.requireNonNull(context.getDatabasePath(DATABASE_NAME).getParent()));
+        fileObserver.startWatching();
+    }
+
+    @After
+    public void teardown() {
+        fileObserver.stopWatching();
+        TestStorageAdapter.cleanup(adapter);
+    }
+
+    /**
+     * Save a record to the database and verify it was saved.
+     * Then call clear and verify that the database file is re-created
+     * and is writable.
+     * @throws DataStoreException bubbles up exceptions thrown from the adapter
+     * @throws IOException if it can't ready file creation time
+     * @throws InterruptedException if something happens when sleeping for 1 second
+     */
+    @Test
+    public void clearDeletesAndRecreatesDatabase() throws DataStoreException, IOException, InterruptedException {
+        assertDbFileExists();
+        assertEquals(0, fileObserver.createFileEventCount);
+        assertEquals(0, fileObserver.deleteFileEventCount);
+        BlogOwner blogger1 = createBlogger("Dummy Blogger Sr.");
+        BlogOwner blogger2 = createBlogger("Dummy Blogger Jr.");
+        //Save a record and check if it's there
+        adapter.save(blogger1);
+        assertRecordIsInDb(blogger1.getName());
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+        assertObserverReceivedRecord(blogger1);
+        //The ensures files are created at least 1 second apart
+        Thread.sleep(1000);
+
+        adapter.clear();
+        //Make sure file was deleted and re-created
+        assertEquals(1, fileObserver.createFileEventCount);
+        assertEquals(1, fileObserver.deleteFileEventCount);
+        assertDbFileExists();
+        //Verify observer is still alive
+        assertFalse(subscriberDisposableRef.get().isDisposed());
+
+        //Make sure the new file is writable
+        adapter.save(blogger2);
+        //Check the new record is in the database
+        //and the old record is not.
+        assertRecordIsInDb(blogger2.getName());
+        assertRecordIsNotInDb(blogger1.getName());
+        assertObserverReceivedRecord(blogger2);
+        //Terminate the adapter
+        adapter.terminate();
+        //Verify observer was disposed.
+        assertTrue(subscriberDisposableRef.get().isDisposed());
+
+    }
+
+    private BlogOwner createBlogger(String name) throws DataStoreException {
+        return BlogOwner.builder()
+            .name(name)
+            .build();
+    }
+
+    private void assertObserverReceivedRecord(BlogOwner blogger) {
+        for (StorageItemChange<? extends Model> owner : observer.values()) {
+            if (owner.item() instanceof BlogOwner &&
```

Comment:
There is also an `item.itemClass()`, you could do like `BlogOwner.class.isAssignableFrom(item.itemClass())` or something like that. (Or, `BlogOwner.class.equals(item.itemClass())`.)

Reply:
I'm going with `BlogOwner.class.isAssignableFrom(item.itemClass())`.